### DISCUSSION TITLE
Cart price update when product price changes

### DISCUSSION THREAD

**Discussion by guillaumeklaus** [swellstores/community#187] [Help] [answered]
Hello all ! 

I am not sure to understand the normal behavior when a product price changes. 

Let’s imagine this following steps : 
- a product is created at $7
- a customer adds this product in its cart 
- I change the product price before the customer converts its cart 

It seems that after those changes, when the cart api is called the grand_total remains unchanged at $7 instead being updated with the new price of the product.

Is it what I should expect ?

**Comment by sags95** [comment, marked as answer]
@guillaumeklaus Yes currently this is the intended behaviour, however there is a feature improvement that's logged in order to update the cart in this case.

**Reply by guillaumeklaus** [reply]
Thanks for the quick answer 👍

**Comment by Raigeku** [comment]
Upping this

Has there been developed any solution to this problem?
Currently, if a user adds an item to the cart whyle at price X, it keeps X price on the cart even after it has been changed.

We tested this on a product on sale: ending the offer does not change the price.

**Reply by ericingram** [reply]
There isn't a standard solution for that, but you could easily implement one using a Swell App function
